File: src/Goods.BackOffice/client-app/src/domain/drivers/driverBlank.ts
import { Gender } from "./driverGender";
import { RightsCategory } from "./driverRightCategory";
import {Driver} from "./driver"
export class DriverBlank {
    constructor(
        public id: string | null,
        public name:string | null,
        public surname:string | null,
        public patronymic: string | null,
        public gender: Gender | null,
        public rightsCategories: RightsCategory[],
        public age: number | null,
        public experience:number | null,
        public transportVechileId: string | null,
        public payment: number | null,
    ) { }
}
export namespace DriverBlank {
    export function getEmpty(): DriverBlank {
        return new DriverBlank(null, null, null, null, null,[],null,null,null,null);
    }

    export function getFromDriver(driver: Driver): DriverBlank {
        return {
            id: driver.id,
            name:driver.name,
            surname:driver.surname,
            patronymic:driver.patronymic,
            gender:driver.gender,
            rightsCategories:driver.rightsCategories,
            age:driver.age,
            experience:driver.experience,
            transportVechileId:driver.transportVechile.id,
            payment:driver.payment
        };
    }
}

File: src/Goods.Domain/Drivers/DriverBlank.cs
using Goods.Domain.Drivers.Enums;
using Goods.Domain.TransportVehicles;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Goods.Domain.Drivers
{
    public class DriverBlank
    {
        public Guid? Id { get; set; } 
        public String? Name { get; set; } 
        public String? Surname { get; set; } 
        public String? Patronymic { get; set; }
        public Gender? Gender { get; set; } 
        public RightsCategory[]? RightsCategories { get; set; }
        public Int32? Age { get; set; } 
        public Int32? Experience { get; set; }
        public Guid? TransportVechileId { get; set; }
        public Double? Payment { get; set; }
    }
}
